### PR TITLE
For JSX text, construct a single literal node `"foo bar"` instead of `"foo" + " " + "bar"`

### PR DESCRIPTION
**Pull request by andy-ms**

### CODE REVIEW

**Comment by DanielRosenwasser** [comment]
Is there an issue tracking this?

**Comment by andy-ms** [comment]
This is not a bug. The output was just unnecessarily complicated.

**Comment by RyanCavanaugh** [comment]
FWIW Babel used to emit the `"foo" + " " + "bar"` but now emits `"foo bar"` as well

**Comment by rbuckton** [comment]
@andy-ms there are some failing tests in this PR.

File: src/compiler/transformers/jsx.ts [hunk 1]
```diff
@@ -151,28 +151,29 @@ namespace ts {
...
-                        const part = text.substr(firstNonWhitespace, lastNonWhitespace - firstNonWhitespace + 1);
-                        if (!parts) {
-                            parts = [];
-                        }
-
-                        // We do not escape the string here as that is handled by the printer
-                        // when it emits the literal. We do, however, need to decode JSX entities.
-                        parts.push(createLiteral(decodeEntities(part)));
+                        const part = decodeEntities(text.substr(firstNonWhitespace, lastNonWhitespace - firstNonWhitespace + 1));
+                        acc = acc === undefined ? part : acc + " " + part;
```
Comment (DanielRosenwasser) [inline src/compiler/transformers/jsx.ts]: If you write

```ts
<div>

    hello\
world
</div>
```

What is the output?

Comment (andy-ms) [inline src/compiler/transformers/jsx.ts]: It will be properly escaped: `"hello\\ world"`.

**Comment by RyanCavanaugh** [comment]
CI's failing again

**Pull request merged**